backend/infrahub/core/convert_object_type/repository_conversion.py

```diff
@@ -0,0 +1,46 @@
+from infrahub import lock
+from infrahub.core.branch import Branch
+from infrahub.core.convert_object_type.object_conversion import InputForDestField, convert_object_type
+from infrahub.core.manager import NodeManager
+from infrahub.core.node import Node
+from infrahub.core.repositories.create_repository import RepositoryFinalizer
+from infrahub.core.schema import NodeSchema
+from infrahub.database import InfrahubDatabase
+
+
+async def convert_repository_type(
+    node: Node,
```

Comment:
I wonder if this should take a union of the protocols for CoreRepository and CoreReadOnlyRepository instead of a Node, that way we should also be able to get rid of the ignore statements for typing. It would also clarify that this function is only for repositories (even if the name also indicates that).

Reply:
Agreed and updated